src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7177,34 +7177,12 @@ private StatementSyntax ParseEmbeddedStatement()
             // The consumers of embedded statements are expecting to receive a non-null statement 
             // yet there are several error conditions that can lead ParseStatementCore to return 
             // null.  When that occurs create an error empty Statement and return it to the caller.
-            StatementSyntax statement = this.ParseStatementCore() ?? SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
-
-            switch (statement.Kind)
-            {
-                // In scripts, stand-alone expression statements may not be followed by semicolons.
-                // ParseExpressionStatement hides the error.
-                // However, embedded expression statements are required to be followed by semicolon. 
-                case SyntaxKind.ExpressionStatement:
-                    if (IsScript)
-                    {
-                        var expressionStatementSyntax = (ExpressionStatementSyntax)statement;
-                        var semicolonToken = expressionStatementSyntax.SemicolonToken;
-
-                        // Do not add a new error if the same error was already added.
-                        if (semicolonToken.IsMissing &&
-                            !semicolonToken.GetDiagnostics().Contains(diagnosticInfo => (ErrorCode)diagnosticInfo.Code == ErrorCode.ERR_SemicolonExpected))
-                        {
-                            semicolonToken = this.AddError(semicolonToken, ErrorCode.ERR_SemicolonExpected);
-                            statement = expressionStatementSyntax.Update(expressionStatementSyntax.Expression, semicolonToken);
-                        }
-                    }
-
-                    break;
-            }
-
-            return statement;
+            return this.ParseStatementCore() ?? ParseEmptyStatement();
```

Comment:
so having a very simple body here causes `ParseEmbeddedStatement` to get inlined? Is this the case in both x86 and x64?

Reply:
i'm not sure how to tell for x86.  but we're passing on x86, so think we're good :)